apps/dashboard/src/components/primitives/field-editor/variable-popover/variable-popover.tsx

```diff
@@ -0,0 +1,257 @@
+import { useState, useRef, useEffect } from 'react';
+import { PopoverContent } from '@/components/primitives/popover';
+import { Input, InputField } from '@/components/primitives/input';
+import { FormControl, FormItem } from '@/components/primitives/form/form';
+import { Switch } from '@/components/primitives/switch';
+import { Code2 } from '../../../icons/code-2';
+import { Separator } from '../../separator';
+import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from '@/components/primitives/select';
+import { TransformerItem } from './components/transformer-item';
+import { TransformerList } from './components/transformer-list';
+import { useVariableParser } from './hooks/use-variable-parser';
+import { useTransformerManager } from './hooks/use-transformer-manager';
+import { formatLiquidVariable } from './utils';
+import type { VariablePopoverProps } from './types';
+
+export function VariablePopover({ variable, onClose, onUpdate }: VariablePopoverProps) {
+  const searchInputRef = useRef<HTMLInputElement>(null);
+  const { parsedName, parsedDefaultValue, parsedTransformers } = useVariableParser(variable);
+  const [name, setName] = useState(parsedName);
+  const [defaultVal, setDefaultVal] = useState(parsedDefaultValue);
+  const [showRawLiquid, setShowRawLiquid] = useState(false);
+  const [searchQuery, setSearchQuery] = useState('');
+
+  useEffect(() => {
+    const handleEscape = (e: KeyboardEvent) => {
```

Comment:
I am surprised we have to handle escape manually. Doesn't the default ShadCN popover close on ESC?